openforcefield/utils/utils.py

```diff
@@ -365,27 +373,40 @@ def convert_all_quantities_to_string(smirnoff_data):
     return obj_to_return
 
 
+@functools.singledispatch
```

Comment:
It's cool to see advanced decorators used properly. This is the sort of feature that's so complex that I've been on the fence about using it, since it may be hard to debug, and could scare potential contributors away. I'm happy to have it in this part of the code as a trial run. 

This could be a bad philosophy for me to have, and I'm open to discussion on it.

Reply:
I can revert to `if...else` if you prefer, no problem. The advantages of the dispatch design is that it's easier to extend (it can even be extended in a different module through `register`) and it's slightly faster. I take it for granted at this point, but I agree this feature is not widely used and we may not need the advantages for this function. I'll leave it up to your decision.

Reply:
Thanks for the details. Let’s keep singledispatch. :-)

Reply:
Ok!